pyuvdata/uvdata/corr_fits_src/corr_fits.pyx

```diff
@@ -37,15 +37,18 @@ cpdef dict input_output_mapping():
 
   return pfb_inputs_to_outputs
 
-cpdef tuple generate_map(
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cpdef void generate_map(
   dict ants_to_pf,
-  dict in_to_out,
-  numpy.ndarray[ndim=1, dtype=numpy.int32_t] map_inds,
-  numpy.ndarray[ndim=1, dtype=numpy.npy_bool] conj
+  numpy.int32_t[:] map_inds,
+  numpy.npy_bool[:] conj,
 ):
   cdef int ant1, ant2, p1, p2, pol_ind, bls_ind, out_ant1, out_ant2
   cdef int out_p1, out_p2, ind1_1, ind1_2, ind2_1, ind2_2, data_index
 
+  cdef dict in_to_out = input_output_mapping()
```

Comment:
Related to your comment on the PR, I think it makes sense to move the mapping call here (in case the compiler can be smart about inlining/calling the function more efficiently). But I would vote for keeping the python function around as well, since it doesn't seem to be harming anything and it would be an API-level change to remove it (so we'd have to go through the deprecation rigmarole).

Reply:
Sounds like a plan to me. I we _could_ turn it into a c++ map too which would make the call even a little more efficient, but I have not really used it before and don't know if there's a good way to go back from a map to a dict. Either way I don't think it is a particularly troublesome part of the code so I have been in favor of leaving it.